linker: add ARC-specific debug sections

tagunil opened the pull request:
ARC architecture has some additional .arcextmap debug sections used by debuggers to understand extension-specific instructions. These sections could safely be ignored, but doing this causes warnings at least with ARC MWDT toolchain.

tejlmand commented:
PTAL @abrodkin @evgeniy-paltsev

evgeniy-paltsev reviewed (changes requested):
As it's arc-architecture specific and these sections are not used for risc-v builds with MWDT toolchain it is worth to move it to ARC linker file instead of adding sections for everyone.

tagunil committed:
linker: arc: add ARC-specific debug sections

ARC architecture has some additional .arcextmap debug sections used by
debuggers to understand extension-specific instructions. These sections
could safely be ignored, but doing this causes warnings at least with
ARC MWDT toolchain.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>
